src/HIE/Bios/Cradle.hs

```diff
@@ -138,14 +138,29 @@ processCabalWrapperArgs args =
             in trace dir $ Just final_args
         _ -> Nothing
 
-cabalAction :: FilePath -> Maybe String -> FilePath -> IO (ExitCode, String, [String])
-cabalAction work_dir mc _fp = do
-  wrapper_fp <- writeSystemTempFile "wrapper.bat" $
-    if isWindows then cabalWrapperBat else cabalWrapper
+-- generate a fake GHC that can be passed to cabal
+-- when run with --interactive, it will print out its
+-- command-line arguments and exit
+getCabalWrapperTool :: IO FilePath
+getCabalWrapperTool = do
+  wrapper_fp <-
+    if isWindows
+      then do
+        wrapper_hs <- writeSystemTempFile "wrapper.hs" cabalWrapperHs
+        -- the initial contents will be overwritten immediately after by ghc
+        wrapper_fp <- writeSystemTempFile "wrapper.exe" ""
+        callProcess "ghc" ["-o", wrapper_fp, wrapper_hs]
+        return wrapper_fp
+      else do
+        writeSystemTempFile "wrapper.bat" cabalWrapper
```

Comment:
We probably shouldn't call it `wrapper.bat` if it's only used on linux?

Reply:
Agreed, I was keeping the old code in the else branch but it seems like an obvious cleanup to change it back.